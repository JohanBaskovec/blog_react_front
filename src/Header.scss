.Header {
  display: flex;
  position: fixed;
  width: 100%;
  top: 0;
  left: 0;
  justify-content: center;
  align-items: center;
  background-color: #53719a;
  color: white;
}

.Header__nav {
  display: flex;
}

.Header__navList {
  margin: 0;
  padding: 0;
  list-style: none;
}

.Header__navListElement {

}
