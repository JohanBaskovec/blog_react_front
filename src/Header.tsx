import "./Header.scss";
import {AppLink} from "./AppLink";

export interface HeaderProps {
    style?: React.CSSProperties;
    contentWidth: string;
}

export function Header(props: HeaderProps) {
    return (<header className="Header" style={props.style}>
        <nav style={{flexBasis: props.contentWidth}}
             className="Header__nav">
            <ul className="Header__navList">
                <li className="Header__navListElement">
                    <AppLink color="white" to="/">Home</AppLink>
                </li>
            </ul>
        </nav>
    </header>)
}
