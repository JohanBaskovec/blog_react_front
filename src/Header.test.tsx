import React from "react";
import TestRenderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect'
import {Header} from "./Header";
import { MemoryRouter } from "react-router-dom";

describe('Header', () => {
    it("is rendered properly", () => {
        // noinspection CheckTagEmptyBody
        const tree = TestRenderer
            .create(
                <MemoryRouter>
                    <Header contentWidth="79%" style={{height: "4px"}}></Header>
                </MemoryRouter>
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
