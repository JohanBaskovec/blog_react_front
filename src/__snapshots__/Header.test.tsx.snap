// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header is rendered properly 1`] = `
<header
  className="Header"
  style={
    Object {
      "height": "4px",
    }
  }
>
  <nav
    className="Header__nav"
    style={
      Object {
        "flexBasis": "79%",
      }
    }
  >
    <ul
      className="Header__left-nav-list"
    >
      <li
        className="Header__navListElement"
      >
        <a
          className="AppLink AppLink--white"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
    </ul>
    <ul
      className="Header__right-nav-list"
    >
      <li
        className="Header__navListElement"
      >
        <a
          className="AppLink AppLink--white"
          href="/login"
          onClick={[Function]}
        >
          Login
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
